project-base/storefront/components/Blocks/Banners/BannersSlider.tsx

```diff
@@ -1,114 +1,116 @@
+        );
+    }, []);
 
     useEffect(() => {
-        const setPauseTrue = () => {
-            setPause(true);
-        };
-        const setPauseFalse = () => {
-            setPause(false);
-        };
-
-        const sliderBox = sliderBoxRef.current;
-
-        if (sliderBox !== null) {
-            sliderBox.addEventListener('mouseover', setPauseTrue);
-            sliderBox.addEventListener('mouseout', setPauseFalse);
+        if (isAutoplayActive) {
+            bannerSlidesRefs?.[activeIndex].current?.scrollIntoView({
+                behavior: 'smooth',
+                block: 'nearest',
+                inline: 'center',
+            });
         }
+    }, [activeIndex, isAutoplayActive]);
 
-        return () => {
-            sliderBox?.removeEventListener('mouseover', setPauseTrue);
-            sliderBox?.removeEventListener('mouseout', setPauseFalse);
-        };
-    }, [sliderRef]);
+    useEffect(() => {
+        if (bannerSlidesRefs && isAutoplayActive && !isHover) {
+            handleSetTimeout();
+        }
+
+        return () => clearTimeout(timeoutFc);
+    }, [isAutoplayActive, bannerSlidesRefs, activeIndex]);
 
     useEffect(() => {
-        timer.current = setInterval(() => {
-            // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-            if (!pause && slider !== null) {
-                slider.next();
-            }
-        }, 5000);
-        return () => {
-            if (timer.current !== null) {
-                clearInterval(timer.current);
-            }
-        };
-    }, [pause, slider]);
-
-    const onMoveToSlideHandler = (newSlideIndex: number) => {
-        slider.moveToSlide(slider.details().absoluteSlide - (currentSlide - newSlideIndex));
-    };
+        let observer: IntersectionObserver;
+
+        if (sliderRef.current) {
+            observer = new IntersectionObserver(
+                ([entry]) => {
+                    if (isAutoplayActive !== entry.isIntersecting) {
+                        setIsAutoplayActive(entry.isIntersecting);
+                    }
+                },
+                {
+                    threshold: [1],
+                    root: null,
+                },
+            );
+
+            observer.observe(sliderRef.current);
+        }
+
+        return () => observer.disconnect();
+    }, [sliderRef, isAutoplayActive]);
 
     return (
-        <div className="flex flex-col gap-6 vl:flex-row" ref={sliderBoxRef} data-testid={dataTestId}>
+        <div
+            className="flex flex-col gap-6 vl:flex-row"
+            data-testid={dataTestId}
+            onMouseEnter={handleHoverEnter}
+            onMouseLeave={handleHoverLeave}
+            onTouchStart={handleHoverEnter}
+            onTouchEnd={handleHoverLeave}
+        >
             <div
+                className={twJoin(
+                    "grid h-80 snap-mandatory rounded-xl [-ms-overflow-style:'none'] [scrollbar-width:'none']",
```

Comment:
Not sure if these styles work as you'd expect them...
[Screencast from 11.8.2023 08:11:46.webm](https://github.com/shopsys/shopsys/assets/55181354/7e0a715c-78e7-44ff-9095-98801f6c3fbd)

Reply:
Yes, Firefox was buggy in some scenarios. If I've fixed it bug in Chrome appeared. So I decided to pull back to previous solution and use this refactoring there.